pages/docs/manual/latest/api/belt/hash-map.mdx

```diff
@@ -31,7 +31,7 @@ let make: (~hintSize: int, ~id: id<'key, 'id>) => t<'key, 'value, 'id>
 `make(~hintSize=10, ~id)` creates a new map by taking in the comparator and `hintSize`.
 
 ```res example
-module IntHash = Belt.Id.MakeHashable({
+module IntHash = Belt.Id.MakeHashableU({
```

Comment:
cc @cristianoc is this needed? This isn't pre-compiled into uncurried via `MakeHashableU`?

Reply:
No the compiler does not come with 2 versions of Belt.
Having to mention a `xxxU` function in documentation is pretty much a red flag, in that it instantaneously introduces tech debt.
There won't be any `xxxU` functions in future, so this is a strange point in time to document their use.

@zth do you have any thoughts about the solution here? The thing is, Belt will beed to be replaced with a library that lives outside the compiler, or in any case alongside RescriptCore. What we are witnessing here is simply that this has not happened yet.

Reply:
The mid to long term idea for Belt is to be a niche lib living outside of the compiler. This is explicit documentation for Belt, and Belt will be much less needed as Core "takes over".

I'm not sure I have a point here, more an observation. I think we can break Belt out to its own package as soon as we've shipped v11. I guess that would alleviate this particular issue.

Another observation is that Belt will be much less prevalent in docs and examples now that Core exists and we're moving everything to that. So this might not be such a big problem going forward as we think.

Reply:
Belt in the compiler can be considered an implementation detail, that can be addressed over time.
I think we can restrict attention to user visible docs for this.

So I guess we can proceed and make a note about next steps in a separate issue.

Reply:
Yes something like this sounds like a good approach.

Though I would try to make the compiler's result, not Belt, the source of truth already now.
As otherwise that's going to pop up somewhere, be it hovering in the editor, or some error message.

Reply:
@cknitt  https://github.com/rescript-lang/rescript-compiler/pull/6450